golem-worker-executor-base/src/services/worker_enumeration.rs

```diff
@@ -0,0 +1,363 @@
+use crate::error::GolemError;
+use crate::services::active_workers::ActiveWorkers;
+use crate::services::golem_config::GolemConfig;
+use crate::services::oplog::{OplogService, OplogServiceDefault};
+use crate::services::worker::{WorkerService, WorkerServiceInMemory, WorkerServiceRedis};
+use crate::services::{golem_config, HasConfig, HasOplogService, HasWorkerService};
+use crate::worker::calculate_last_known_status;
+use crate::workerctx::WorkerCtx;
+use async_trait::async_trait;
+use golem_common::model::{TemplateId, WorkerFilter, WorkerId, WorkerMetadata, WorkerStatus};
+use golem_common::redis::RedisPool;
+use std::sync::Arc;
+use tracing::info;
+
+#[async_trait]
+pub trait RunningWorkerEnumerationService {
+    async fn get(
+        &self,
+        template_id: &TemplateId,
+        filter: Option<WorkerFilter>,
+    ) -> Result<Vec<WorkerMetadata>, GolemError>;
+}
+
+#[derive(Clone)]
+pub struct RunningWorkerEnumerationServiceDefault<Ctx: WorkerCtx> {
+    active_workers: Arc<ActiveWorkers<Ctx>>,
+}
+
+#[async_trait]
+impl<Ctx: WorkerCtx> RunningWorkerEnumerationService
+    for crate::services::worker_enumeration::RunningWorkerEnumerationServiceDefault<Ctx>
+{
+    async fn get(
+        &self,
+        template_id: &TemplateId,
+        filter: Option<WorkerFilter>,
+    ) -> Result<Vec<WorkerMetadata>, GolemError> {
+        info!(
+            "Get workers for template: {}, filter: {}",
+            template_id,
+            filter.is_some()
+        );
+
+        let active_workers = self.active_workers.enum_workers();
+
+        let mut template_workers: Vec<WorkerMetadata> = vec![];
+        for worker in active_workers {
+            if worker.0.template_id == *template_id
+                && (worker.1.metadata.last_known_status.status == WorkerStatus::Running
+                    || worker.1.metadata.last_known_status.status == WorkerStatus::Idle)
```

Comment:
I think it should only be `Running`. For idle ones we cannot use this method because it does not return all the ones that are idle - just the ones that are in memory. On the other hand there cannot be anything that is running and is not in memory.